libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/DownloadUpdateListener.kt

```diff
@@ -0,0 +1,45 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+import com.mapbox.services.android.navigation.v5.utils.DownloadTask
+import java.io.File
+
+/**
+ * Triggers a [TileUnpacker] to unpack the tar file into routing tiles once the FILE_EXTENSION_TAR
+ * download is complete.
+ */
+internal class DownloadUpdateListener(
+    private val downloader: RouteTileDownloader,
+    private val tileUnpacker: TileUnpacker,
+    tilePath: String,
+    tileVersion: String,
+    private val listener: RouteTileDownloadListener
+) : DownloadTask.DownloadListener {
+
+    private val destinationPath: String
+
+    companion object {
```

Comment:
Should we extract this constant to a file constant and get rid of the `companion object`?

Reply:
Look suggestion above.

Reply:
it's strong recommendation: for constant  use `companion object`. 
1. `const` inlines to code where it's used.
2. if we move `val DOWNLOAD_ERROR_MESSAGE` to class then it stays property and in java bytecode it looks like `filed` + `getField()` method

Reply:
> 1. const inlines to code where it's used.

Couldn't that be more performant? I also think the compiler does that behind the scenes with all constants.

> 2. if we move val DOWNLOAD_ERROR_MESSAGE to class then it stays property and in java bytecode it looks like filed + getField() method

Couldn't that be workarounded / 💅 using `@JvmStatic` annotations?

Reply:
1. No, it's actually created property with 'getter' and in java it looks like 'getValue(): String'
2. @JvmStatic only for `companion object` and `Object` 😉